src/xbot/common/properties/DatabaseStorageBase.java

```diff
@@ -13,25 +13,140 @@
+            log.error(e.toString());
+            e.printStackTrace();
+        } catch (Exception e) {
+            log.error("Ran into a general expection, so could not open a connection to the database! No properties will be loaded or persisted!!");
+            log.error(e.toString());
+            e.printStackTrace();
+        }        
+        
+        // create properties if not exists
+        if (propertiesTableExists() == false)
+        {
+            createPropertiesTable();
+        }
+    }
+    
+    public void setDouble(String key, double value){
+        saveProperty("double", key, Double.toString(value));
+    }
+    
+    public void setBoolean(String key, boolean value){
+        saveProperty("boolean", key, Boolean.toString(value));
+    }
+    
+    public void setString(String key, String value){
+        saveProperty("string", key, value);
+    }
+    
+    public Double getDouble(String key){
+        String value = loadProperty(key);
+        try {
+            return Double.parseDouble(value);
+        }
+        catch (NumberFormatException e)
+        {
+            return null;
+        }
+    }
+    
+    public Boolean getBoolean (String key){
+        String value = loadProperty(key);
+        return parseBoolean(value);
+    }
+    
+    private Boolean parseBoolean(String value)
+    {
+        if (value.toLowerCase().equals("true"))
+        {
+            return true;
+        }
+        
+        if (value.toLowerCase().equals("false"))
+        {
+            return false;
+        }
+        
+        return null;
+    }
+    
+    public String getString (String key){
+        String value = loadProperty(key);
+        if (value.length() == 0)
+        {
+            return null;
+        }
+        return value;
+    }
+    
+    public void clear(){
+        obliterateStorage();
+    }
+    
+    private void saveProperty(String type, String name, String value) {
+       
+        String payload = "UPDATE PROPERTIES SET TYPE='" + name + "', VALUE='" + type + "' WHERE NAME = '" + value + "'";
+        Statement sta;
+        try {
+            sta = conn.createStatement();
+            int count = sta.executeUpdate(payload);
+
+            if (count == 0) {
+                // Looks like this isn't currently in the database. We need to add it instead.
+                payload = "INSERT INTO PROPERTIES VALUES ('" + name + "', '" + type + "', '" + value + "')";
+                Statement insert = conn.createStatement();
+                count = insert.executeUpdate(payload);
+            }
+        } catch (SQLException e) {
+            log.warn("Unable to save property " + name + "!");
+            log.warn(e.toString());
+            e.printStackTrace();
+        }
+    }
+    
+    private String loadProperty(String name)
+    {
+        try {
+            Statement sta = conn.createStatement();
+            String payload = "SELECT * FROM PROPERTIES WHERE NAME = '" + name + "'";
```

Comment:
I'll see if, during table creation, I can easily define the "names" column as having to be unique.